packages/jsii-pacmak/test/expected.jsii-calc-base-of-base/kotlin/src/main/kotlin/software/amazon/jsii/tests/calculator/baseofbase/VeryBaseProps.kt

```diff
@@ -0,0 +1,46 @@
+package software.amazon.jsii.tests.calculator.baseofbase
+
+@javax.annotation.Generated("jsii-pacmak")
+interface VeryBaseProps : software.amazon.jsii.JsiiSerializable {
+    val foo: software.amazon.jsii.tests.calculator.baseofbase.Very
+
+    class Builder {
```

Comment:
be cool if we could do this in a more Kotlin DSLy kinda way, I'm thinking you could add something like this to the interface:

```kt
companion object {
    operator fun invoke(block: Builder.() -> Unit): VeryBaseProps {
        return Builder().apply(block).build()
    }
}
```

So then you can use it like so:

```kt
val props = VeryBaseProps {
    foo = Very.create()
}
```

Reply:
Considering Kotlin's combo of named arguments and default parameter values, I feel there should be no need for a Builder pattern in Kotlin. In fact, I'd see it as a main advantage of a Kotlin impl to abandon the Builder pattern. A constructor for a props object with all the required parameters up front and the ones if default values following should function identically to a Builder at the call site, right?

Reply:
The main issues I see with this approach is backwards compatibility, when we're generating code - how do we know what was previously required vs not? It's currently legal for service teams to drop "required" on proprieties so the property order may change. There's nothing to force a consumer to explicitly name parameters at the call site. If the order of a bunch of `String` properties changes then consumers are going to be in for a bad time.

By going the "DSL Builder" route we force explicit names to properties.